friendly-captcha/modules/profile-builder/profile_builder_reset_password.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+add_filter( 'wppb_recover_password_generate_password_input', 'frcaptcha_pb_reset_password_show_widget', 10, 1);
+
+function frcaptcha_pb_reset_password_show_widget($output) {
+    $plugin = FriendlyCaptcha_Plugin::$instance;
+    if (!$plugin->is_configured() or !$plugin->get_pb_reset_password_active()) {
+        return $output;
+    }
+
+    frcaptcha_enqueue_widget_scripts();
+
+    $widget = frcaptcha_generate_widget_tag_from_plugin($plugin);
+    return str_replace('</ul>', '<li class="wppb-form-field wppb-recaptcha">' . $widget . '</li>' . '</ul>', $output);
+}
+
+add_filter('wppb_recover_password_sent_message1', 'frcaptcha_pb_reset_password_sent_message', 10, 1);
+
+function frcaptcha_pb_reset_password_sent_message($message) {
+//    We are using the plugins ReCaptcha functionality to display the error
+    $plugin = FriendlyCaptcha_Plugin::$instance;
+    if (!$plugin->is_configured() or !$plugin->get_pb_reset_password_active()) {
+        return $message;
+    }
+    $solution = frcaptcha_get_sanitized_frcaptcha_solution_from_post();
+
+    if ( empty( $solution ) ) {
+        $message = 'wppb_recaptcha_error';
+    }
+
+    $verification = frcaptcha_verify_captcha_solution($solution, $plugin->get_sitekey(), $plugin->get_api_key());
+
+    if (!$verification['success']) {
+        $message = 'wppb_recaptcha_error';
+    }
+
+    return $message;
+}
+
+
+add_filter('wppb_recover_password_displayed_message1','frcaptcha_pb_reset_password_display_message', 10, 1);
+
+function frcaptcha_pb_reset_password_display_message($message){
+//    This will only be triggered when the captcha error occurred, so we can safely send this error.
+//    We can't check again because then we will get an error from FriendlyCaptcha
+    return $message = $message . '<p class="wppb-warning">' . FriendlyCaptcha_Plugin::default_error_user_message() . '</p>';
+}
```

Comment:
Isn't this called for any error message that is displayed and not just captcha errors?

Reply:
I looked at the code again and you are absolutely right! There are specific situations where this error could be displayed. The original plugin just sends another request to the ReCaptcha API, but doing that with the FC API threw an error when I tried to do the same thing. Do you possibly know of any way to make another call to the API? Otherwise I would try to sort out the problem in a different way.

Reply:
Looking at the source code of the plugin (https://github.com/wp-plugins/profile-builder/blob/7454755ea7016c597ac9f279c91f3930805a6f4e/assets/misc/plugin-compatibilities.php#L163-L194) again I think what you did is actually fine. The sole purpose of the filter seems to be to get the right error message from recaptcha 🤔

Reply:
> Looking at the source code of the plugin (https://github.com/wp-plugins/profile-builder/blob/7454755ea7016c597ac9f279c91f3930805a6f4e/assets/misc/plugin-compatibilities.php#L163-L194) again I think what you did is actually fine. The sole purpose of the filter seems to be to get the right error message from recaptcha 🤔

Yes, the inner workings of the plugin are pretty weird (especially the ReCaptcha functionality). But then it seems like my first thoughts were right.

Reply:
Thanks again for the PR! This will be part of the next version :)